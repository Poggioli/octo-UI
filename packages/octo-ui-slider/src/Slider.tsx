import { forwardRef } from "@octo-ui/core";
import {
  StyledSliderRange,
  StyledSliderRoot,
  StyledSliderThumb,
  StyledSliderTrack,
} from "./styles";
import {
  SliderProps,
  SliderRangeProps,
  SliderThumbProps,
  SliderTrackProps,
} from "./types";

const SliderThumb = forwardRef<typeof StyledSliderThumb, SliderThumbProps>(
  ({ ...props }, forwardedRef) => {
    return <StyledSliderThumb {...props} ref={forwardedRef} />;
  }
);
SliderThumb.toString = () => `.${StyledSliderThumb.className}`;

// ========================================================================= //

const SliderRange = forwardRef<typeof StyledSliderRange, SliderRangeProps>(
  ({ ...props }, forwardedRef) => {
    return <StyledSliderRange {...props} ref={forwardedRef} />;
  }
);
SliderRange.toString = () => `.${StyledSliderRange.className}`;

// ========================================================================= //

const SliderTrack = forwardRef<typeof StyledSliderTrack, SliderTrackProps>(
  ({ ...props }, forwardedRef) => {
    return <StyledSliderTrack {...props} ref={forwardedRef} />;
  }
);
SliderTrack.toString = () => `.${StyledSliderTrack.className}`;

// ========================================================================= //

const Slider = forwardRef<typeof StyledSliderRoot, SliderProps>(
  ({ thumbs = 1, ...props }, forwardedRef) => {
    return (
      <StyledSliderRoot {...props} ref={forwardedRef}>
        <SliderTrack>
          <SliderRange />
        </SliderTrack>
        {Array.from({ length: thumbs }).map((_, index) => (
          <SliderThumb key={index} />
        ))}
      </StyledSliderRoot>
    );
  }
);
Slider.toString = () => `.${StyledSliderRoot.className}`;

export { SliderThumb, SliderRange, SliderTrack, Slider };
export {
  StyledSliderThumb,
  StyledSliderRange,
  StyledSliderTrack,
  StyledSliderRoot,
};
