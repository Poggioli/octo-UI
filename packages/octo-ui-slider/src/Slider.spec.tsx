import { render } from "@testing-library/react";
import { Slider, SliderRange, SliderThumb, SliderTrack } from "./Slider";

describe("<Slider />", () => {
  it(`GIVEN a <Slider />
      WHEN render is called
      THEN should render a <Slider />`, () => {
    const rendered = render(<Slider />);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Slider />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Slider.toString()).toBe(".octo-UI-c-eWmfpo");
  });

  it(`GIVEN a <SliderThumb />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SliderThumb.toString()).toBe(".octo-UI-c-iAtlAa");
  });

  it(`GIVEN a <SliderTrack />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SliderTrack.toString()).toBe(".octo-UI-c-gbhTzR");
  });

  it(`GIVEN a <SliderRange />
      WHEN toString is called
      THEN return a string`, () => {
    expect(SliderRange.toString()).toBe(".octo-UI-c-eFZlhW");
  });
});
