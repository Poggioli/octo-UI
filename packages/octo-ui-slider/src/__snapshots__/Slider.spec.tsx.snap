// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> GIVEN a <Slider />
      WHEN render is called
      THEN should render a <Slider /> 1`] = `
<div>
  <span
    aria-disabled="false"
    class="octo-UI-c-eWmfpo octo-UI-c-eWmfpo-RICCB-color-unstyled"
    data-orientation="horizontal"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="octo-UI-c-gbhTzR"
      data-orientation="horizontal"
    >
      <span
        class="octo-UI-c-eFZlhW"
        data-orientation="horizontal"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="octo-UI-c-iAtlAa"
        data-orientation="horizontal"
        data-radix-collection-item=""
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</div>
`;
